src/replication.c

```diff
@@ -3390,3 +3450,265 @@ void replicationStartPendingFork(void) {
         }
     }
 }
+
+/* Find replica at IP:PORT from replica list */
+static client *findReplica(char *host, int port) {
+    listIter li;
+    listNode *ln;
+    client *replica;
+
+    listRewind(server.slaves,&li);
+    while((ln = listNext(&li))) {
+        replica = ln->value;
+        char ip[NET_IP_STR_LEN], *replicaip = replica->slave_ip;
+
+        if (replicaip[0] == '\0') {
+            if (connPeerToString(replica->conn, ip, sizeof(ip), NULL) == -1)
+                continue;
+            replicaip = ip;
+        }
+
+        if (!strcasecmp(host, replicaip) &&
+                (port == replica->slave_listening_port))
+            return replica;
+    }
+
+    return NULL;
+}
+
+const char *getFailoverStateString() {
```

Comment:
Maybe expose the rest of the failover state? Attaching to an instance stuck in failover that can be valuable debugging info I believe.

Reply:
I thought about this, so when we are in the failover state, our master/host values will be set and timeout isn't valid. That should be enough to debug what is going on. If we are in the sync state, the data is available through our offset and the slaves section, and I'm not sure if adding timeout is that useful.

We can add more data here, but I think there is enough to investigate right now.